src/Core/Vault/Services/Implementations/AzureAttachmentStorageService.cs

```diff
@@ -251,16 +251,17 @@ private async Task DeleteAttachmentsForPathAsync(string path)
 
     private async Task InitAsync(string containerName)
     {
-        if (!_attachmentContainers.ContainsKey(containerName) || _attachmentContainers[containerName] == null)
+        if (!_attachmentContainers.TryGetValue(containerName, out var attachmentContainer) || attachmentContainer == null)
         {
-            _attachmentContainers[containerName] = _blobServiceClient.GetBlobContainerClient(containerName);
+            attachmentContainer = _blobServiceClient.GetBlobContainerClient(containerName);
+            _attachmentContainers[containerName] = attachmentContainer;
```

Comment:
In the existing code, it is already added before the creation.

If you are worried about concurrency issues, the `_attachmentContainers` `Dictionary` should also be changed to be a `ConcurrentDictionary` to avoid [concurrency issues](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0#thread-safety)

If concurrency should be supported, I suggest doing it in a separate pull request